pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -306,20 +307,32 @@ func (policy *regoEnforcer) getReasonNotAllowed(enforcementPoint string, input i
 		return fmt.Errorf("%s not allowed by policy. Input unavailable due to marshalling error", enforcementPoint)
 	}
 
+	defaultMessage := fmt.Errorf("%s not allowed by policy. Security policy is not valid. Please check security policy or re-generate with tooling. Input: %s", enforcementPoint, string(inputJSON))
+
 	input["rule"] = enforcementPoint
 	result, err := policy.rego.Query("data.policy.reason", input)
-	if err == nil {
-		errors, _ := result.Value("errors")
-		if errors != nil {
-			if len(errors.([]interface{})) > 0 {
-				return fmt.Errorf("%s not allowed by policy. Errors: %v. Input: %s", enforcementPoint, errors, string(inputJSON))
-			} else {
-				return fmt.Errorf("%s not allowed by policy. Security policy is not valid. Please check security policy or re-generate with tooling. Input: %s", enforcementPoint, string(inputJSON))
-			}
-		}
+	if err != nil {
+		return defaultMessage
```

Comment:
is there a reason the error value is ignored here (and in the original code too)?

Reply:
An error here means that, when queried about why the policy denied the request, it produced an error. This could be for a few different reasons:

1. The policy is a custom policy that doesn't define the "reason" rule, and so there is no further information to provide
2. The policy has an error that only occurs when the reason rule in queried

In either case, the error returned here is an error in the error message system, and appending that to the error message can get very confusing with the current error messages. With the upcoming change to JSON error messages we can separate out this category of error in a way that will cause less confusion, so this may be a good thing to consider there.

Reply:
can we add the log statement? or at the least add a `//TODO:` comment to annotate the decision?